cpp/src/Slice/Parser.cpp

```diff
@@ -2255,16 +2365,16 @@ Slice::Container::hasClassDefs() const
 bool
 Slice::Container::hasInterfaceDefs() const
 {
-    for(ContainedList::const_iterator p = _contents.begin(); p != _contents.end(); ++p)
+    for (const auto& p : _contents)
     {
-        ClassDefPtr cl = ClassDefPtr::dynamicCast(*p);
-        if(cl && cl->isInterface())
+        InterfaceDefPtr def = InterfaceDefPtr::dynamicCast(p);
+        if (def)
```

Comment:
Like we do for the `hasClassDefs` function:
```suggestion
        if (InterfaceDefPtr::dynamicCast(p))
```

Reply:
Fixed